### PR TITLE
Device provides logger

### PR DESCRIPTION
**Pull request by Grigory-Rylov**

### CODE REVIEW

File: src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java [hunk 1]
```diff
@@ -158,19 +159,16 @@ public synchronized void installPackage(String absolutePath, boolean reinstall,
...
      */
     public synchronized void pullCoverageFile(InstrumentalExtension instrumentationInfo,
                                               String coverageFilePrefix,
                                               String coverageFile,
-                                              File outCoverageDir,
-                                              final ILogger logger) throws PullCoverageException {
+                                              File outCoverageDir) throws PullCoverageException {
         MultiLineReceiver outputReceiver = new MultilineLoggerReceiver(logger);
 
         // TODO: Why another logger is used here?
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java:170]: not needed anymore

File: src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java [hunk 2]
```diff
@@ -158,19 +159,16 @@ public synchronized void installPackage(String absolutePath, boolean reinstall,
...
-        logger.verbose("ConnectedDeviceWrapper '%s': fetching coverage data from %s",
-                getName(), coverageFile);
+        logger.i(TAG,"Fetching coverage data from %s", coverageFile);
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java:171]: nit: space after `,`

File: src/main/java/com/github/grishberg/tests/DeviceCommandsRunner.java [hunk 3]
```diff
@@ -31,32 +31,30 @@ boolean runCommands(List<ConnectedDeviceWrapper> devices, final TestRunnerContex
...
                     }
                 } catch (Throwable e) {
-                    logger.e(threadTag, "Execute command exception:", e);
+                    logger.e(TAG, "Execute command exception:", e);
                     commandException = e;
                 } finally {
                     deviceCounter.countDown();
                 }
-                logger.i(threadTag, "Thread {} ({}) is finished", tid, device);
+                logger.i(TAG, "Command execution thread is finished");
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/DeviceCommandsRunner.java:57]: nit: there is actually no `thread` implications anymore, it's all about `device tasks finished` now or something

File: src/main/java/com/github/grishberg/tests/common/DeviceSpecificLogger.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        logger.e(createTag(tag), message, throwable);
+    }
+
+    override fun w(tag: String, msgFormat: String, vararg args: Any?) {
+        logger.w(createTag(tag), msgFormat, args)
+    }
+
+    private fun createTag(tag: String): String = "${device.name} / $tag"
+
+}
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/common/DeviceSpecificLogger.kt:45]: no new line

File: src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java [hunk 5]
```diff
@@ -224,16 +220,16 @@ public synchronized String executeShellCommandAndReturnOutput(String command) th
...
-        MultilineLoggerReceiver(ILogger logger) {
+        MultilineLoggerReceiver(RunnerLogger logger) {
             this.logger = logger;
         }
 
         @Override
         public void processNewLines(String[] lines) {
             for (String line : lines) {
-                logger.verbose(line);
+                logger.i(TAG, line);
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java:232]: why `i` here and not `v`?

Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java:232]: there is no any method `v`

Comment (Nexen23) [reply src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java:232]: @Nirvanchik won't it flood logs with uneeded data?

Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java:232]: I can use `d` in this place

Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java]: changed to `d`

**Comment by Grigory-Rylov** [comment]
Take logger from device in `DeviceCommandRunner`

File: src/test/java/com/github/grishberg/tests/InstrumentationTestLauncherTest.java [hunk 6]
```diff
@@ -52,7 +52,7 @@ public InstrumentationTestLauncherTest() {
     public void setUp() throws Exception {
         List<ConnectedDeviceWrapper> devicesList = Arrays.asList(deviceWrapper);
         when((adbWrapper.provideDevices())).thenReturn(devicesList);
-        when(deviceCommandsRunnerFabric.provideDeviceCommandRunner(eq(logger), any(DeviceRunnerCommandProvider.class))).thenReturn(runner);
+        when(deviceCommandsRunnerFabric.provideDeviceCommandRunner(any(DeviceRunnerCommandProvider.class))).thenReturn(runner);
```
Comment (Nirvanchik) [inline src/test/java/com/github/grishberg/tests/InstrumentationTestLauncherTest.java:55]: тут отступ не поехал случайно?

Comment (Nirvanchik) [reply src/test/java/com/github/grishberg/tests/InstrumentationTestLauncherTest.java:55]: Блин, в Overview норм отображает, в широком диффе - криво (((

File: src/main/java/com/github/grishberg/tests/common/DeviceSpecificLogger.kt [hunk 7]
```diff
@@ -0,0 +1,44 @@
+package com.github.grishberg.tests.common
+
+import com.android.ddmlib.IDevice
+
+/**
+ * [RunnerLogger] wrapper with device name as TAG postfix.
```
[the remark this answers is not in the document]
Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/common/DeviceSpecificLogger.kt:6]: ох, уже да, изначально ж. был постфикс

**Pull request merged**